### ISSUE TITLE
Image transposing

### ISSUE THREAD

**Issue by kbwestfall** [PYPIT/PYPIT#386]
From past experience, I'm not a huge fan of transposing arrays just because of the row-major vs. column-major conflict between fits writers and python fits readers.

I did this for the MaNGA DAP.  The cubes are written in IDL as (x, y, lambda), but python reads and returns the cubes as (lambda, y, x).  Highly annoying.  So one of the first things I do in the DAP is transpose all the data arrays.  This is one of my biggest regrets because now reading the data takes far longer than it should.

For pypit (and what pypit will generally deal with), this may not be an issue, but I wanted to bring this up just because of my experience of doing the same thing and then regretting it later.

**Comment by rcooke-ast** [comment]
That's very interesting! Originally, PypIt didn't transpose arrays that were input. This quickly became a huge pain since different instruments disperse the light along a row or a column. We ended up having a bunch of cython code that was effectively duplicate (i.e. `if dispaxis=0: foo else bar`), which made it very difficult to maintain (we would invariably have two identical pieces of code, except for the dispaxis. Sometimes the code associated with dispaxis=0 was modified and not the other).

So, we decided to bite the bullet and transpose everything from the beginning, to avoid the need of duplicate code that was difficult to maintain.

If it's just a matter of speed, I think we can deal with this using hardware (or multiprocess the transposing action).
